rct229/rulesets/ashrae9012019/section1/section1rule5.py

```diff
@@ -0,0 +1,203 @@
+from rct229.utils.jsonpath_utils import find_one
+
+APPLICABLE_LIMIT = 0.05
+
+
+class Section1Rule5(RuleDefinitionBase):
+    """Rule 5 of ASHRAE 90.1-2019 Appendix G Section 1 (Performance Calculations)"""
+
+    def __init__(self):
+        super(Section1Rule5, self).__init__(
+            rmrs_used=produce_ruleset_model_instance(
+                USER=True,
+                BASELINE_0=True,
+                BASELINE_90=True,
+                BASELINE_180=True,
+                BASELINE_270=True,
+                PROPOSED=True,
+            ),
+            rmrs_used_optional=produce_ruleset_model_instance(
+                BASELINE_90=True,
+                BASELINE_180=True,
+                BASELINE_270=True,
+            ),
+            id="1-5",
+            description="When on-site renewable energy generation exceeds the thresholds defined in Section 4.2.1.1, the methodology defined in this section shall be used to calculate the PCI.",
+            ruleset_section_title="Performance Calculations",
+            standard_section="Section 4.2.1.1",
+            is_primary_rule=True,
+            rmr_context="ruleset_model_descriptions/0",
+        )
+
+    def is_applicable(self, context, data=None):
+        rmd_u = context.USER
+        rmd_b0 = context.BASELINE_0
+        rmd_b90 = context.BASELINE_90
+        rmd_b180 = context.BASELINE_180
+        rmd_b270 = context.BASELINE_270
+        rmd_p = context.PROPOSED
+
+        pbp_set = []
+        bbp_set = []
+        pbp_nre_set = []
+        for rmd in (rmd_u, rmd_b0, rmd_b90, rmd_b180, rmd_b270, rmd_p):
+            if rmd is not None:
+                pbp_set.append(
+                    find_one(
+                        "$.output.total_proposed_building_energy_cost_including_renewable_energy",
+                        rmd,
+                    )
+                )
+                bbp_set.append(
+                    find_one(
+                        "$.output.baseline_building_performance_energy_cost",
+                        rmd,
+                    )
+                )
+                pbp_nre_set.append(
+                    find_one(
+                        "$.output.total_proposed_building_energy_cost_excluding_renewable_energy",
+                        rmd,
+                    )
+                )
+
+        pbp_set = list(set(filter(lambda x: x is not None, pbp_set)))
+        bbp_set = list(set(filter(lambda x: x is not None, bbp_set)))
+        pbp_nre_set = list(set(filter(lambda x: x is not None, pbp_nre_set)))
+
+        assert_(len(pbp_set) >= 1, "At least one `pbp_set` value must exist.")
+        assert_(len(bbp_set) >= 1, "At least one `bbp_set` value must exist.")
+        assert_(len(pbp_nre_set) >= 1, "At least one `pbp_nre_set` value must exist.")
+
+        return (
+            len(pbp_set) == len(bbp_set) == len(pbp_nre_set) == 1
+            and bbp_set[0] != 0
+            and (pbp_nre_set[0] - pbp_set[0]) / bbp_set[0] > APPLICABLE_LIMIT
```

Comment:
Base on the applicability check, I think the first condition is not needed.
This is because the failure of the first condition should result in a failure condition, not an NOT_APPLICABLE.

Similarly, for the `bbp_set[0] != 0`, you can put an `assert_` around it to result in UNDETERMINED outcome instead of `NOT_APPLICABLE`

Reply:
Good point. Addressed.